quickwit/quickwit-doc-mapper/src/default_doc_mapper/field_mapping_entry.rs

```diff
@@ -649,6 +723,13 @@ fn deserialize_mapping_type(
             }
             return Ok(FieldMappingType::Object(object_options));
         }
+        QuickwitFieldType::Concatenate => {
+            let concatenate_options: QuickwitConcatenateOptions = serde_json::from_value(json)?;
+            if concatenate_options.concatenate_fields.is_empty() {
+                anyhow::bail!("concatenate type must have at least one sub-field");
```

Comment:
if we do the dynamic property thingy, we need to remove this check.

Reply:
I've made the check take into account `include_dynamic_fields`